tests/nn/misc/test_checkpoint_activations_norm.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+# pylint: disable=missing-module-docstring
+# pylint: disable=missing-class-docstring
+# pylint: disable=missing-function-docstring
+
+""" Test checkpoint_wrapper with normalization layers. """
+
+import pytest
+import torch
+from torch.nn import BatchNorm2d, LayerNorm, Linear, Sequential
+from torch.optim import SGD
+
+from fairscale.nn.misc.checkpoint_activations import checkpoint_wrapper
+from fairscale.utils.testing import objects_are_equal, torch_version
+
+NORM_TYPES = [LayerNorm, BatchNorm2d]
+MP_TYPES = ["fp32", "fp16", "call_half"]
+
+
+def get_model(norm_type, checkpointed, mixed_precision):
+    assert norm_type in NORM_TYPES, norm_type
+    assert checkpointed in [True, False], checkpointed
+    assert mixed_precision in MP_TYPES
+
+    model = Sequential(Linear(3, 2), norm_type(2))
+
+    if mixed_precision == "fp16":
+        # Set param.data and buffers as fp16
+        for p in model.parameters():
+            p.data = p.data.half()
+        for m in model:
+            for n, b in m.named_buffers():
+                setattr(b, n, b.half())
```

Comment:
shouldn't this be `setattr(m, n, b.half())`?

Reply:
yes! what's a dumb bug!